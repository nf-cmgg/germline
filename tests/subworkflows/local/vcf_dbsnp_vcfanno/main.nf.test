nextflow_workflow {

    name "Test Workflow VCF_DBSNP_VCFANNO"
    script "subworkflows/local/vcf_dbsnp_vcfanno/main.nf"
    workflow "VCF_DBSNP_VCFANNO"

    tag "subworkflows"
    tag "subworkflows_local"
    tag "vcf_dbsnp_vcfanno"

    test("vcf_dbsnp_vcfanno - default") {

        when {
            params {
                annotate = true
            }
            workflow {
                """
                input[0] = Channel.of([
                    [id:"NA24143", family:"NA24143", family_samples:"NA24143", caller:"haplotypecaller"],
                    file(params.vcf1, checkIfExists:true),
                    file(params.tbi1, checkIfExists:true)
                ])
                input[1] = Channel.value([
                    [id:"dbnsp"],
                    file(params.vcf2, checkIfExists:true)
                ])
                input[2] = Channel.value([
                    [id:"dbnsp"],
                    file(params.tbi2, checkIfExists:true)
                ])
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    workflow.out.vcfs.collect { [ it[0], "${file(it[1]).name},variantsMD5:${path(it[1]).vcf.variantsMD5}", file(it[2]).name ] }
                ).match() }
            )
        }

    }

}
