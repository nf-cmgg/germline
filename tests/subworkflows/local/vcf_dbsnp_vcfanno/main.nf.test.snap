{
    "vcf_dbsnp_vcfanno - default": {
        "content": [
            [
                [
                    {
                        "id": "NA24143",
                        "family": "NA24143",
                        "family_samples": "NA24143",
                        "caller": "haplotypecaller"
                    },
                    "NA24143.vcf.gz,variantsMD5:b4f76bc67ba0e159489393d4788349b3",
                    "NA24143.vcf.gz.tbi"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.1",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-11-05T14:44:22.532988474"
    }
}